packages/@aws-cdk/aws-synthetics/test/canary.test.ts

```diff
@@ -0,0 +1,223 @@
+import '@aws-cdk/assert/jest';
+import { objectLike } from '@aws-cdk/assert';
+import * as iam from '@aws-cdk/aws-iam';
+import * as s3 from '@aws-cdk/aws-s3';
+import { App, Duration, Lazy, Stack } from '@aws-cdk/core';
+import * as synthetics from '../lib';
+
+let stack: Stack;
+beforeEach(() => {
+  stack = new Stack(new App(), 'canaries');
+});
+
+test('Basic canary properties work', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'mycanary',
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+    successRetentionPeriod: Duration.days(10),
+    failureRetentionPeriod: Duration.days(10),
+    startAfterCreation: false,
+    timeToLive: Duration.minutes(30),
+    runtime: synthetics.Runtime.SYNTHETICS_1_0,
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'mycanary',
+    SuccessRetentionPeriod: 10,
+    FailureRetentionPeriod: 10,
+    StartCanaryAfterCreation: false,
+    Schedule: objectLike({ DurationInSeconds: '1800'}),
+    RuntimeVersion: 'syn-1.0',
+  });
+});
+
+test('Canary can have generated name', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
```

Comment:
Suggestion: use a placeholder that is somewhat more descriptive of the real value folks should use (don't have to make it look like a function, but for example: `/* Synthetics handler code */`).

Remember people might look at tests for a source of examples of how the module is supposed to be used.